### PR TITLE
[stable/concourse] credhub credential manager options

### LINKED ISSUE #32: Configuration endpoint for Credhub secrets provider
For credentials management, I'd like a configuration endpoint for credhub.

Although there doesn't appear to be a helmchart to deploy credhub on Kubernetes, I'd like to configure our Concourse helm chart to leverage our existing credhub deployed via BOSH running within the same VPC.

[Currently there are `values.yml` endpoints for vault, awsSsm, and awsSecretsManager](https://github.com/helm/charts/blob/master/stable/concourse/values.yaml#L305); could one be opened up for Credhub as well?

### PR DESCRIPTION
**Pull request by aegershman**
#### What this PR does / why we need it:

Adds Credhub as a credentials manager. Currently we have most of the "official" credentials providers implemented as options except Credhub.

#### Which issue this PR fixes
  - fixes concourse/hush-house#32

#### Special notes for your reviewer:

- This implementation allows Concourse to talk to [Credhub via `client_id` and `client_secret` ](https://github.com/concourse/concourse/blob/4030b93c761cdddaa4e0c5748a9234ee64a11917/atc/creds/credhub/manager.go#L33-L36). I'm under the impression you _could_ do [mTLS an auth model(see also: `CONCOURSE_CREDHUB_CLIENT_CERT`, `CONCOURSE_CREDHUB_CLIENT_KEY`)](https://github.com/concourse/concourse/blob/4030b93c761cdddaa4e0c5748a9234ee64a11917/atc/creds/credhub/manager.go#L26-L31)? ... But ultimately I decided to stick to what's [included in the doc, which is a CA_Cert and `client-id/client-secret`](https://concourse-ci.org/credhub-credential-manager.html)

- I did local testing to the best of my abilities && tried to keep the scope focused. Would appreciate an honest overview/review of this work; I’m by no means a `helm` pro or have deep experience with this chart. **If this doesn't meet quality standards && someone else wants to implement this as they see fit, rock and roll, please do.**. But having Credhub be an option would be massively helpful for my team.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by aegershman** [comment]
Apologies if I'm doing this wrong: /cc @cirocosta @YoussB

**Comment by aegershman** [comment]
Apologies, not trying to pester, but considering [there's PR for Concourse 5.3.0 flags which](https://github.com/helm/charts/pull/14292) bumps the chart's minor version to 6.3.0, I hope someone from the Concourse team also takes a look at this PR as it likely also warrant a minor version bump.

File: stable/concourse/templates/web-secrets.yaml [hunk 1]
```diff
@@ -71,6 +71,11 @@ data:
   vault-client-key: {{ default "" .Values.secrets.vaultClientKey | b64enc | quote }}
   vault-client-auth-param: {{ default "" .Values.secrets.vaultAuthParam | b64enc | quote }}
   {{- end }}
+  {{- if .Values.concourse.web.credhub.enabled }}
+  credhub-ca-cert: {{ default "" .Values.secrets.credhubCaCert | b64enc | quote }}
+  credhub-client-id: {{ default "" .Values.secrets.credhubClientId | b64enc | quote }}
```
Comment (cirocosta) [inline stable/concourse/templates/web-secrets.yaml]: Hey, if these are actually required, I think we might be able to do something similar as we do for `oidc` 🤔  wdyt??

**Comment by cirocosta** [comment]
Hi @aegershman , thanks for the PR!!

Nothing wrong at all 😁  

I added some comments to the changes, please let me know what you think 👍 

Thanks!

File: stable/concourse/templates/web-secrets.yaml (hunk 1 above)
Comment (aegershman) [reply stable/concourse/templates/web-secrets.yaml]: Sure thing, give me just a bit.

**Comment by aegershman** [comment]
@cirocosta Now that the Chart version is at 6.3.0, is it appropriate to bump to 6.4.0 or 6.3.1?

**Comment by cirocosta** [comment]
Thanks for following up with the comments, @aegershman !

I'd say `6.4.0` as specifying that'd be adding some new functionality that couldn't be done before assuming that the `values.yaml` declares the interface by which the semantic versioning applies to.

Wdyt?

thanks!

**Comment by aegershman** [comment]
Not a problem @cirocosta thanks for the time, I appreciate it.

I've updated the Chart to be `6.4.0` 👍 let me know if there's anything else I can do to help.

**Comment by cirocosta** [comment]
Thanks, @aegershman !

I think the bump in the image name could fit in a separate PR (where we'd be performing a patch then).

Wdyt?

thanks!

**Comment by aegershman** [comment]
Alrighty @cirocosta sounds good, will leave this PR as-is. Let me know if there's anything else you need me to do to help get this merged in 👍 thanks a ton (cc @dashaun)

**Review by cirocosta — APPROVED** [review]

**Comment by aegershman** [comment]
Thank you!

**Pull request merged**